code/mission/missionlog.cpp

```diff
@@ -479,14 +335,11 @@ void mission_log_add_entry_multi( LogType type, const char *pname, const char *s
 
 int mission_log_get_time_indexed( LogType type, const char *pname, const char *sname, int count, fix *time)
 {
-	int i, found;
-	log_entry *entry;
 	Assertion(count > 0, "The count parameter is %d; it should be greater than 0!", count);
 
-	entry = &log_entries[0];
-
-	for (i = 0; i < last_entry; i++, entry++) {
-		found = 0;
+	for (const auto &ii: Log_entries) {
+		auto entry = &ii;
+		bool found = false;
```

Comment:
Changing found to a bool, I think assignments to found should also assign bool literals instead of ints as it's currently the case.

Reply:
Yeah.  Originally I wanted to minimize the size of the diff but I'll put this in a new commit.